tensorflow/c/kernels.h

```diff
@@ -65,6 +66,11 @@ typedef struct TF_KernelBuilder TF_KernelBuilder;
 typedef struct TF_OpKernelConstruction TF_OpKernelConstruction;
 typedef struct TF_OpKernelContext TF_OpKernelContext;
 
+// TF_InitKernel to do op/kernel registration.
+// Plugin should either implement TF_InitKernel to register kernels or use
+// static registration. This function should register all kernels in a plugin.
```

Comment:
Why would the plugin choose one over the other?

Reply:
This comment is from @annarev ,  I think one is existing custom op through C++ api and one is C api?

Reply:
Originally we were aiming to just support static registration, as it is done here:
https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/c/kernels/bitcast_op.cc;l=167;drc=16cab3e8ce1d988b81fbcf07c18e4e097706d29f

@jzhoulon added `TF_InitKernel` which I think is even better than static registration. I don't know if we want to deprecate the original static registration approach at this point or if there is a disadvantage to having both ways.

Reply:
I'd prefer removing static registration.  We can always add the ability to use static registration later (right?), but once it is there we'll have a harder time removing it.   (We should also add a check that device registration only happens inside `TF_InitKernel`.)

Reply:
Removing "static registration" mention from the comment sounds good to me.